online-store/src/components/basket/buyForm.ts

```diff
@@ -0,0 +1,174 @@
+                        value="rrrrr rrrrr rrrrr"
+                        >
+                    </div>
+                </div>
+                    <div class="col">
+                    <h3 class="title">payment</h3>
+                    <div class="inputBox">
+                        <span>cards accepted:</span>
+                        <div class="imageBox">
+                        <span class="inputBox__imgVisa"></span>
+                        <span class="inputBox__imgMaster"></span>
+                        <span class="inputBox__imgUnion"></span>
+                        </div>
+                    </div>
+                    <div class="inputBox">
+                        <span>card number:</span>
+                        <input class="input js-input js-input-card-number" name="card-number"
+                        type="text" placeholder="1111222233334444"
+                        required
+                        pattern="[0-9]{16}"
+                        title="16 цифр"
+                        value="1111111111111111">
+                    </div>
+                    <div class="flex">
+                        <div class="inputBox">
+                            <span>valid thru:</span>
+                            <input class="input js-input js-input-card-term" name="card-term"
+                            type="text" placeholder="mm/yy"
+                            required
+                            pattern="^(0[1-9]|1[0-2])\\/(0[1-9]|1[0-9]|2[0-9]|3[0-1])$"
+                            title="Срок действия в формате mm/yy"
+                            >
+                        </div>
+                        <div class="inputBox">
+                            <span>CVV:</span>
+                            <input class="input js-input js-input-card-cvv" name="card-cvv"
+                            type="text" placeholder="123"
+                            required
+                            pattern="[0-9]{3}"
+                            title="3 цифры"
+                            value="123">
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <input type="submit" value="buy" class="submit-btn">
+        </form>
+  `;
+  buyForm.insertAdjacentHTML('afterbegin', buyFormHTML);
+
+  // Outside buyForm-close
+  shadowWrapper.addEventListener('click', shadowWrapperCloser);
+  function shadowWrapperCloser(): void {
+    shadowWrapper.classList.add('hide');
+    buyForm.classList.add('hide');
+    body.classList.remove('overflow-hidden');
+  }
+
+  // Form card-term validation
+  const inputCardNumber = document.querySelector(
+    '.js-input-card-number'
+  ) as HTMLInputElement;
+  const inputCardTerm = document.querySelector(
+    '.js-input-card-term'
+  ) as HTMLInputElement;
+
+  inputCardTerm.addEventListener('input', () => {
+    validCardTerm();
+  });
+
+  function validCardTerm(): void {
```

Comment:
this is probably a very large function that does a lot of things. It is worth avoiding this. Large functions are very complex in the future